pkg/controller/registry/containerregistry.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+	"google.golang.org/api/storage/v1"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+
+	"time"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/crossplane/provider-gcp/apis/registry/v1alpha1"
+	gcp "github.com/crossplane/provider-gcp/pkg/clients"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// Error strings.
+const (
+	errNewStorageClient = "cannot create new Google Storage Client"
+	errNotGcr           = "managed resource is not a Google Container Registry (GCR)"
+	errHandshake        = "an error occurred during handshake"
+	errGetBucket        = "cannot get Bucket object"
+	errCreate           = "cannot create Google Container Registry (GCR) object"
+	errUpdate           = "cannot update Google Container Registry (GCR) object"
+
+	gcrUrl                          = "gcr.io"
+	testRepoName                    = "crossplane-test"
```

Comment:
When we try to handshake with `"gcr.io/crossplane-playgroud"` the following error occurs:

> message: 'create failed: an error occurred during handshake: GET https://eu.gcr.io/v2/token?scope=repository%3Acrossplane-playground%3Apush%2Cpull&service=eu.gcr.io: 
UNKNOWN: Pushing to root-level images is disabled. See https://cloud.google.com/container-registry/docs/support/deprecation-notices'

So we need to add a repository name to the end of URL: "gcr.io/crossplane-playgroud/crossplane-test"

Also, please see the Terraform side (`my-repo`): https://github.com/hashicorp/terraform-provider-google/blob/9795b29a7b2157fb82287bafa4880063ebbefcba/google/resource_container_registry.go#L67